bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java

```diff
@@ -0,0 +1,242 @@
+        }
+        // audio volatility
+        param = newConfig.get(PARAM_AUDIOVOLATITLITY);
+        if (param != null) {
+            config.audioVolatility = Double.parseDouble(param.toString());
+        }
+
+        // phoneme volatility
+        param = newConfig.get(PARAM_PHONEMEVOLATITLITY);
+        if (param != null) {
+            config.phonemeVolatility = Double.parseDouble(param.toString());
+        }
+
+        // speakingRate
+        param = newConfig.get(PARAM_SPEAKINGRATE);
+        if (param != null) {
+            config.speakingRate = Double.parseDouble(param.toString());
+        }
+    }
+
+    @Override
+    public String getId() {
+        return SERVICE_ID;
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    public Set<Voice> getAvailableVoices() {
+        String url = config.url + LIST_VOICES_URL;
+        try {
+            String responseVoices = HttpRequestBuilder.getFrom(url).getContentAsString();
+            VoiceDto[] mimicVoiceResponse = gson.fromJson(responseVoices, VoiceDto[].class);
+            if (mimicVoiceResponse == null) {
+                logger.error("Cannot get mimic voices from the URL {}", url);
+                return Collections.emptySet();
+            }
+            if (mimicVoiceResponse.length == 0) {
+                logger.warn("Voice set from mimic is empty ?!");
+            }
+            Set<Voice> voiceSet = new HashSet<>();
+            for (VoiceDto voiceDto : mimicVoiceResponse) {
+                if (voiceDto.speakers != null && voiceDto.speakers.size() > 0) {
+                    for (String speaker : voiceDto.speakers) {
+                        voiceSet.add(new MimicVoice(voiceDto.key, voiceDto.language, voiceDto.name, speaker));
+                    }
+                } else {
+                    voiceSet.add(new MimicVoice(voiceDto.key, voiceDto.language, voiceDto.name, null));
+                }
+            }
+            return voiceSet;
+        } catch (IOException e) {
+            logger.error("Cannot get mimic voices from the URL {}", url);
+            return Collections.emptySet();
+        }
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return Set.<AudioFormat> of(AUDIO_FORMAT);
+    }
+
+    /**
+     * Checks parameters and calls the API to synthesize voice.
+     *
+     * @param text Input text.
+     * @param voice Selected voice.
+     * @param requestedFormat Format that is supported by the target sink as well.
+     * @return Output audio stream
+     * @throws TTSException in case the service is unavailable or a parameter is invalid.
+     */
+    @Override
+    public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFormat) throws TTSException {
+        logger.debug("Synthesize '{}' for voice '{}' in format {}", text, voice.getUID(), requestedFormat);
+        // Validate arguments
```

Comment:
You are not checking that the voice is valid, meaning an available Mimic voice.

Reply:
Indeed. I added a check.